File: hackerrank.com/problem-solving/time_conversion.py
# https://www.hackerrank.com/challenges/time-conversion/problem

def timeConversion(s):
    hour, min, half = s[0:2], s[2:8], s[-2:-1]
    if half == 'A':
        newHour = '00' if hour == '12' else hour
    else:
        newHour = hour if hour == '12' else str(int(hour) + 12)
    return newHour + min


if __name__ == '__main__':
    s = input()
    print(timeConversion(s))
    

File: hackerrank.com/problem-solving/time_conversion.js
// https://www.hackerrank.com/challenges/time-conversion/problem

function timeConversion(s) {
  var match = s.match(/(\d{2})(:\d{2}:\d{2})(A|P)M/);
  var hour = parseInt(match[1]);
  if (match[3] === 'A' && hour === 12) return `00${match[2]}`;
  if (match[3] === 'A') return `${match[1]}${match[2]}`
  if (hour === 12) return `${match[1]}${match[2]}`
  return `${hour + 12}${match[2]}`
}


function main(inputData) {
  const lines = inputData.split("\n");
  // const n = parseInt(lines[0])
  const s = lines[0]
  // const ar = lines[1].split(' ').map(x => parseInt(x))
  console.log(timeConversion(s))
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
let _input = '';
process.stdin.on("data", input => {
  _input += input;
});

process.stdin.on("end", () => {
  main(_input);
});
